Bottom: 024e8f23f30fce49d9f002733af596f3ef6a8cf5
Top:    2e502adf7fcf5f816133c2c68f8fcd3216b86671
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-06 01:08:36 +0100

action.c: In ActionLoadFrom(), only hide the crosshair where required


---

diff --git a/src/action.c b/src/action.c
index 65a3c3a..1380d54 100644
--- a/src/action.c
+++ b/src/action.c
@@ -5919,18 +5919,20 @@ ActionLoadFrom (int argc, char **argv, int x, int y)
   function = argv[0];
   name = argv[1];
 
-  HideCrosshair ();
-
   if (strcasecmp (function, "ElementToBuffer") == 0)
     {
+      HideCrosshair ();
       if (LoadElementToBuffer (PASTEBUFFER, name, true))
 	SetMode (PASTEBUFFER_MODE);
+      RestoreCrosshair ();
     }
 
   else if (strcasecmp (function, "LayoutToBuffer") == 0)
     {
+      HideCrosshair ();
       if (LoadLayoutToBuffer (PASTEBUFFER, name))
 	SetMode (PASTEBUFFER_MODE);
+      RestoreCrosshair ();
     }
 
   else if (strcasecmp (function, "Layout") == 0)
@@ -5957,7 +5959,6 @@ ActionLoadFrom (int argc, char **argv, int x, int y)
       LoadPCB (fname);
     }
 
-  RestoreCrosshair ();
   return 0;
 }
